src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java

```diff
@@ -265,12 +259,22 @@
             consumes = APPLICATION_JSON_UTF8_VALUE,
             produces = APPLICATION_JSON_UTF8_VALUE
     )
+
     @ResponseBody
-    public ResponseEntity<CreateUserProfileResponse> updateUserProfile(@Valid @RequestBody UpdateUserProfileData updateUserProfileData, @PathVariable String userId) {
+    public ResponseEntity<UserProfileRolesResponse> updateUserProfile(@Valid @RequestBody UpdateUserProfileData updateUserProfileData, @PathVariable String userId
+    ) {
         log.info("Updating user profile");
+        UserProfileRolesResponse userProfileResponse = new UserProfileRolesResponse();
+        if (StringUtils.isEmpty(updateUserProfileData.getRolesAdd())) {
 
-        userProfileService.update(updateUserProfileData, userId);
-        return ResponseEntity.status(HttpStatus.OK).build();
+            log.info("Updating user profile without roles");
+            userProfileService.update(updateUserProfileData, userId);
+        } else {
```

Comment:
Why are you using a validator but not returning a boolean? It looks like validateUserProfileDataAndUserId is doing more than validating and manipulating the data somehow. Please rename.

Reply:
The validator just throws a missing field exception if there is invalid data in the request. There doesn't seem to be any reason to return a boolean

Reply:
See below comment. You need to try to code in a conventional way, which means making it easy for the reviewer to understand what it is you are doing. So either return a boolean and explicitly throw a Runtime exception preferably in the same layer where the exception is occuring. See example below.
If you have a lot of validation logic wrapped in a validation object, fine. But flat is better than nested, so doing something really critical like throwing a runtime exception needs to be made explicit and not obfuscated deep inside some validation logic with a poorly named method that gives me no indication that any exception is being thrown.